march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -106,22 +106,25 @@ march::Joint HardwareBuilder::createJoint(const YAML::Node& joint_config, const
     mode = march::ActuationMode(joint_config["actuationMode"].as<std::string>());
   }
 
-  auto imc =
-      HardwareBuilder::createIMotionCube(joint_config["imotioncube"], mode, urdf_joint, pdo_interface, sdo_interface);
-  if (!imc)
+  std::shared_ptr<march::MotorController> controller;
+  if (joint_config["imotioncube"])
   {
-    ROS_WARN("Joint %s does not have a configuration for an IMotionCube", joint_name.c_str());
+    controller = this->createIMotionCube(joint_config["imotioncube"], mode, urdf_joint, pdo_interface, sdo_interface);
+  }
+  if (!controller)
+  {
+    ROS_FATAL("Joint %s does not have a configuration for a motor controller", joint_name.c_str());
   }
 
-  auto ges = HardwareBuilder::createTemperatureGES(joint_config["temperatureges"], pdo_interface, sdo_interface);
+  auto ges = this->createTemperatureGES(joint_config["temperatureges"], pdo_interface, sdo_interface);
   if (!ges)
   {
     ROS_WARN("Joint %s does not have a configuration for a TemperatureGes", joint_name.c_str());
   }
-  return { joint_name, net_number, allow_actuation, std::move(imc), std::move(ges) };
+  return { joint_name, net_number, allow_actuation, std::move(controller), std::move(ges) };
 }
 
-std::unique_ptr<march::IMotionCube> HardwareBuilder::createIMotionCube(const YAML::Node& imc_config,
+std::shared_ptr<march::IMotionCube> HardwareBuilder::createIMotionCube(const YAML::Node& imc_config,
```

Comment:
Why has this been changed to a `shared_ptr`?

Reply:
Same